datafusion/sql/src/expr/function.rs

```diff
@@ -349,13 +350,31 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                     .map(|e| self.sql_expr_to_logical_expr(*e, schema, planner_context))
                     .transpose()?
                     .map(Box::new);
-                return Ok(Expr::AggregateFunction(expr::AggregateFunction::new_udf(
-                    fm,
+
+                let raw_aggregate_function = RawAggregateUDF {
+                    udf,
                     args,
                     distinct,
                     filter,
                     order_by,
                     null_treatment,
+                };
+
+                for planner in self.planners.iter() {
```

Comment:
this means that any planner extensions would have precidence over the built in aggregate functions from the Registry. I think that is ok

Reply:
We can avoid clone like this

```rust
for planner in self.planners.iter() {
                    match planner.plan_aggregate_udf(raw_aggregate_function)? {
                        PlannerResult::Planned(e) => return Ok(e),
                        PlannerResult::Original(e) => {
                            raw_aggregate_function = e;
                        }
                    }
                }

                return Ok(Expr::AggregateFunction(expr::AggregateFunction::new_udf(
                    raw_aggregate_function.udf,
                    raw_aggregate_function.args,
                    raw_aggregate_function.distinct,
                    raw_aggregate_function.filter,
                    raw_aggregate_function.order_by,
                    raw_aggregate_function.null_treatment,
                )));
```